apps/meteor/ee/app/livechat-enterprise/server/business-hour/Multiple.ts

```diff
@@ -132,6 +132,27 @@ export class MultipleBusinessHoursBehavior extends AbstractBusinessHourBehavior
 		return this.handleRemoveAgentsFromDepartments(deletedDepartment, agentsIds, options);
 	}
 
+	async onDepartmentDisabled(department: ILivechatDepartment): Promise<void> {
+		if (!department.businessHourId) {
+			return;
+		}
+		// Get business hour
+		const businessHour = await this.BusinessHourRepository.findOneById(department.businessHourId);
+		if (!businessHour) {
+			return;
+		}
+		// Check if i'm the only department on this business hour
+		const imTheOnlyOne = !(await LivechatDepartment.countByBusinessHourId(businessHour._id));
```

Comment:
2 points here:
1) Shouldn't we be checking enabled departments only over here?
2) I'm a bit confused if this works. Let's say there's only one dep connected to BH and it's the one which we're disabling right now, then "imTheOnlyOne" would be (!1) => false. On the other hand, if there are more deps connected to this BH, then "imTheOnlyOne" would be "(!2)" => false. Due to this "closeBusinessHour" never gets executed. Please let me know if I'm missing something here 😅

Reply:
For 1: no, we want all. There's another job that will remove all the disabled departments whenever the server starts + business hours is enabled, so in here we can be safe we won't count disabled ones.

For 2: :kek: forgot